packages/babel-generator/src/printer.ts

```diff
@@ -119,7 +119,16 @@ class Printer {
   declare _inputMap: TraceMap;
 
   declare format: Format;
-  inForStatementInitCounter: number = 0;
+
+  inForStatementInit: boolean = false;
+  enterForStatementInit(val: boolean) {
+    const old = this.inForStatementInit;
+    if (old === val) return () => {};
```

Comment:
Maybe return undefined and then use `exit?.()` on the caller?

Reply:
Isn't it better to use consistent types, rather than having polymorphic functions? It's trivial for an engine to optimise an empty function call.